domain/schema/model/sql/0022-resource.sql

```diff
@@ -50,15 +45,30 @@ CREATE TABLE resource (
     CONSTRAINT fk_resource_state_id
     FOREIGN KEY (state_id)
     REFERENCES resource_state (id),
-    CONSTRAINT fk_resource_hash_type_id
-    FOREIGN KEY (hash_type_id)
-    REFERENCES object_store_metadata_hash_type (id)
+    CONSTRAINT fk_resource_store_uuid
+    FOREIGN KEY (resource_store_uuid)
+    REFERENCES resource_store(uuid)
+    CONSTRAINT fk_resource_supplied_by_uuid
+    FOREIGN KEY (supplied_by_uuid)
+    REFERENCES resource_supplied_by (uuid)
 );
 
-CREATE UNIQUE INDEX idx_resource ON resource (application_uuid, name, state_id);
+CREATE VIEW v_resource AS
+SELECT * FROM resource
+JOIN application_resource ON resource.uuid = application_resource.resource_uuid;
```

Comment:
Most of the time we'll want available resources, not ones added by the charmrevisionupdater worker to say what upgrades are available. Consider adding:
```
INNER JOIN resource_state ON resource_state.id = resource.state_id
WHERE resource_state.name = 'available'
```

Reply:
removed view - add when the pattern becomes apparent.